### PR TITLE
 feat: improve readability of middleware handling

### PR DESCRIPTION
**Pull request by aearly**

### CODE REVIEW

File: lib/middleware-runner.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      return inner(...iterArgs)
+    }
+
+    return middlewareFn(...iterArgs, once(iter))
+  }
+
+  return iter(...args, once(iter))
+}
+
+function once (fn) {
```
Comment (aearly) [inline lib/middleware-runner.js]: The `once` package does a lot more that we need it to, like assigning props on the function and binding to a context, likely slowing things down.

File: lib/server.js [hunk 2]
```diff
@@ -121,38 +120,35 @@ module.exports = class Server {
       this.server.once('error', reject)
     })
 
-    this.processServerOnion = onion.sprout(
+    this.processServerMWStack = middlewareRunner(
```
Comment (aearly) [inline lib/server.js]: I hope this is a better name. :D

File: lib/middleware-runner.js (hunk 1 above)
Comment (aearly) [reply lib/middleware-runner.js]: FWIW, middleware functions are already bound here: https://github.com/npm/spife/blob/master/lib/server.js#L97

**Comment by max-b** [comment]
Thanks @aearly !

**Pull request merged**